src/generate/rust.rs

```diff
@@ -64,7 +64,7 @@ impl<Pat: PartialEq> RuleWithNamedFields<Pat> {
                 .collect();
             for (field, paths) in &self.fields {
                 for path in paths {
-                    if path.len() == 0 {
```

Comment:
I think this should be using slice pattern-matching on `path` (if that's a thing on stable), instead of checking the length and using `path[0]`.
The patterns would be `[]`, `[variant]` and `[variant, ..]`, I think?